### PR TITLE
fix: Comic Shanns issues, new fonts structure

### PR DESCRIPTION
**Pull request by Mrazator**
Fixes https://github.com/excalidraw/excalidraw/issues/8327

Adopted https://github.com/jesusmgg/comic-shanns-mono with few modifications:
- cleared unnecessary characters for box drawings and braille
- added polish Ł & ł from https://github.com/kaBeech/serious-shanns/tree/main
- corrected `tumed question mark` glyph

Also done:
- unified prod font paths across `excalidraw-app`, OSS package (host apps, examples, E+), and CDN into `fonts/[FontFamilyName]/[FontFamilyName]-*.woff2`
  - grouping into subfolders per font family ✅
  - unification of `EXCALIDRAW_ASSET_PATH` for OSS and E+ ✅ (instead of one relying on "/" while the other on "/fonts/")
  - unified and organized `excalidraw-app/build` & `excalidraw/dist/prod/` into `fonts/` folder ✅ (instead of having all on root)
    Screenshot 2024-10-15 at 23 30 11
  - host apps can just copy the whole "fonts" folder instead of all the woff2 files ✅
- split all non-deprecated fonts into multiple chunks (atm only Excalifont and Comic Shanns left) and related:
  - stopped adding vite `-[hash]` suffix to fonts on `excalidraw-app` build (we have hashes thanks to the split)
  - optimized fetch during export to force cache
- stopped preloading Virgil, as at this point it's unnecessary for new scenes, while old scenes will mostly get it from the cache or just fetch it on init if needed

Fixes:
- #8677
- #8678 

Todo:
- [ ] Update new font structure to the DO @dwelle

### CODE REVIEW

File: scripts/buildPackage.js [hunk 1]
```diff
@@ -157,5 +158,7 @@ const createESMRawBuild = async () => {
   await buildProd(rawConfigChunks);
 };
 
-createESMRawBuild();
-createESMBrowserBuild();
+(async () => {
+  await createESMRawBuild();
+  await createESMBrowserBuild();
+})();
```
Comment (dwelle) [inline scripts/buildPackage.js:164]: btw node.js should wait for this before script exists, but not sure if guaranteed. Should we make this into an `mjs` script and use top-level await?

File: packages/excalidraw/scene/export.ts [hunk 2]
```diff
@@ -541,7 +541,10 @@ function* fontFacesIterator(
             return;
           }
 
-          const fontFaceOrder = familyIndex + fontFaceIndex;
+          const fontFaceOrder = Number(
+            familyIndex.toString() + fontFaceIndex.toString(),
+          );
```
Comment (dwelle) [inline packages/excalidraw/scene/export.ts:546]: this is potentially buggy as well as you _could_ have `"1" + "0"` and `"0" + "10"` or combinations, so we'll need to use something like cantor pairing or similar:

`(a: number, b: number) => ((a + b) * (a + b + 1)) / 2 + b`

Comment (Mrazator) [reply packages/excalidraw/scene/export.ts:546]: yep, I had there a buffer pair family before, not sure where it disappear, will re-introduce

File: scripts/buildPackage.js (hunk 1 above)
Comment (Mrazator) [reply scripts/buildPackage.js]: actually it should not be really needed looking at the change, just added it defensively as the build was failing from time to time; will revert and we'll see what happens

**Review by dwelle — APPROVED** [review]

**Comment by dwelle** [comment]
Thanks, we're good to go!

**Pull request merged**